src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,240 @@
+        Topic currentAwsRegion = currentdeviceConfiguration.getAWSRegion();
+        Topic newAwsRegion = deploymentConfiguration.getAWSRegion();
+        return !Topic.compareValue(currentAwsRegion, newAwsRegion);
+    }
+
+    private boolean mqttHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        Topics currentMqtt = currentdeviceConfiguration.getMQTTNamespace();
+        Topics newMqtt = deploymentConfiguration.getMQTTNamespace();
+        return !Topics.compareChildren(currentMqtt, newMqtt);
+    }
+
+    private boolean iotDataEndpointHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentIotDataEndpoint = currentdeviceConfiguration.getIotDataEndpoint();
+        Topic newIotDataEndpoint = deploymentConfiguration.getIotDataEndpoint();
+        return !Topic.compareValue(currentIotDataEndpoint, newIotDataEndpoint);
+    }
+
+    private boolean greengrassDataPlaneEndpointHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentGreengrassDataPlaneEndpoint = currentdeviceConfiguration.getGGDataEndpoint();
+        Topic newGreengrassDataPlaneEndpoint = deploymentConfiguration.getGGDataEndpoint();
+        return !Topic.compareValue(currentGreengrassDataPlaneEndpoint, newGreengrassDataPlaneEndpoint);
+    }
+
+    private boolean greengrassDataPlanePortHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        Topic currentGreengrassDataPlanePort = currentdeviceConfiguration.getGreengrassDataPlanePort();
+        Topic newGreengrassDataPlanePort = deploymentConfiguration.getGreengrassDataPlanePort();
+        return !Topic.compareValue(currentGreengrassDataPlanePort, newGreengrassDataPlanePort);
+    }
+
+    /**
+     * Creates an MQTT client and checks if the device can connect to AWS.
+     *
+     * @param totallyCompleteFuture Future will be updated if validation fails
+     */
+    @SuppressWarnings("PMD.AvoidInstanceofChecksInCatchClause")
+    public boolean mqttClientCanConnect(CompletableFuture<DeploymentResult> totallyCompleteFuture) {
+        logger.atDebug().log("Checking MQTT client can connect");
+
+        MqttClientConnection connection = null;
+        String message;
+        try (AwsIotMqttConnectionBuilder builder = mqttClient.createMqttConnectionBuilder(deploymentConfiguration,
+                securityService, null)) {
+            String clientId = Coerce.toString(deploymentConfiguration.getThingName()) + clientIdSuffix;
+            connection = builder.withClientId(clientId).build();
+            int operationTimeoutMillis = MqttClient.getMqttOperationTimeoutMillis(deploymentConfiguration);
+            connection.connect().get(operationTimeoutMillis, TimeUnit.MILLISECONDS);
+            return true;
+        } catch (MqttException | TimeoutException | ExecutionException e) {
+            if (e instanceof MqttException) {
```

Comment:
please try and avoid an if else ladder
rather, break this catch down into multiple catches like
```
catch (MqttException e) {
    do something
} catch (TimeoutException e) {
    do something else
} catch (ExecutionException e) {
    do something else again
}
```

Reply:
also, generally, ExecutionException is a wrapper of the actual cause of the failure
instead of getting the message from execution exception, you should get the message of the exception which is wrapped inside the execution exception using `e.getCause`